### PR TITLE
Convert unit tests to RSpec (from Minitest)

### PR DESCRIPTION
**Pull request by erran-r7**
This enables #104 by using a testing framework standard within the Rapid7 (internal/open source) community.

cc @bcall-r7

### CODE REVIEW

**Comment by bcall-r7** [comment]
Ship it!

File: spec/convection/model/template/resource/loggroups_spec.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    subject do
+      template_json
+    end
+
+    it 'template is defined' do
+      expect(subject).to_not eq(nil)
+    end
+
+    it 'resource values are properly set' do
+      expect(subject['Resources'].include?('testgroup'))
```
Comment (erran-r7) [inline spec/convection/model/template/resource/loggroups_spec.rb]: @bcall-r7 this should be:

```ruby
expect(subject['Resources']).to have_key('testgroup')
```

File: spec/convection/model/template/resource/iam_role_spec.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        .fetch('Properties')
+        .fetch('AssumeRolePolicyDocument')
+    end
+
+    it 'AssumeRolePolicyDocument is not nil' do
+      expect(subject).to_not eq(nil)
+    end
+
+    it 'the policy statement documents principal is bar.amazonaws.com' do
+      expect(subject['Statement'].include?('bar.amazonaws.com'))
```
Comment (erran-r7) [inline spec/convection/model/template/resource/iam_role_spec.rb]: Update this to actually assert something: should be "expect(...).to[_not] ..."

File: spec/convection/model/template/template_spec.rb [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    it 'template has key Mappings' do
+      expect(subject['Mappings']).to be_a(Hash)
+    end
+
+    it 'template has key Conditions' do
+      expect(subject['Conditions']).to be_a(Hash)
+    end
+
+    it 'template has key Resources' do
+      expect(subject['Resources']).to be_a(Hash)
```
Comment (erran-r7) [inline spec/convection/model/template/template_spec.rb]: These "has key" tests should all be converted to the following:

```ruby
# okay
it 'template has key Resources' do
  expect(subject['Resources']).to be_a(Hash)
end

# better - use the implicit subject RSpec syntax
it { is_expected.to have_key('Resources') }
```

**Comment by erran-r7** [comment]
:shipit: on @bcall-r7's specs after my comments are addressed.

File: spec/convection/model/template/resource/iam_role_spec.rb [hunk 4]
```diff
@@ -25,7 +25,7 @@ class Convection::Model::Template::Resource
     end
 
     it 'the policy statement documents principal is bar.amazonaws.com' do
-      expect(subject['Statement'].include?('bar.amazonaws.com'))
+      expect(subject['Statement'][0]['Principal']).to have_value('bar.amazonaws.com')
```
Comment (erran-r7) [inline spec/convection/model/template/resource/iam_role_spec.rb]: Here you could do:

```ruby
expect(subject['Statement']).to include(a_hash_containing('Principal' => 'bar.amazonaws.com'))
```

Comment (bcall-r7) [inline spec/convection/model/template/resource/iam_role_spec.rb]: Nice, i like that.

**Pull request merged**